src/js/feedback/reducers/index.js

```diff
@@ -0,0 +1,78 @@
+import {
+  REVEAL_FORM,
+  SET_FORM_VALUES,
+  SEND_FEEDBACK,
+  FEEDBACK_RECEIVED,
+  FEEDBACK_ERROR,
+  CLEAR_FEEDBACK_ERROR
+} from '../actions';
+
+const initialState = {
+  requestPending: false,
+  feedbackReceived: false,
+  errorMessage: null,
+  formIsVisible: false,
+  formIsSubmittable: false,
+  formValues: {
+    description: '',
+    email: '',
+    shouldSendResponse: false
+  },
+  formErrors: {
+    description: '',
+    email: ''
+  }
+};
+
+function feedbackReducer(state = initialState, action) {
+  switch (action.type) {
+    case REVEAL_FORM:
+      return {
+        ...state,
+        formIsVisible: true
+      };
+    case SET_FORM_VALUES:
+    {
+      const formValues = { ...state.formValues, ...action.formValues };
+      const formErrors = { ...state.formErrors, ...action.formErrors };
+
+      let formIsSubmittable = true;
```

Comment:
Honestly, it's fine like it is. It could be clearer by explicitly splitting
```js
const mapStateToProps = (state) => state.feedback;
```
into
```js
const mapStateToProps = (state) => ({
  formIsSubmittable: state.feedback.formIsSubmittable
  // Etc.
});
```
It seems kinda dumb to do that when you want the whole `state.feedback`, though, so I wouldn't worry about it. I was just being dense. :man_shrugging: